src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Windows.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class NativeLibrary
+    {
+        private static IntPtr LoadLibraryHelper(string libraryName, int flags, ref LoadLibErrorTracker errorTracker)
+        {
+            IntPtr ret = Interop.Kernel32.LoadLibraryEx(libraryName, IntPtr.Zero, flags);
+            if (ret != IntPtr.Zero)
+            {
+                return ret;
+            }
+
+            int lastError = Marshal.GetLastWin32Error();
+            if (lastError != Interop.Errors.ERROR_INVALID_PARAMETER)
+            {
+                errorTracker.TrackErrorCode(lastError);
+            }
+
+            return ret;
+        }
+
+        private static void FreeLib(IntPtr handle)
+        {
+            Debug.Assert(handle != IntPtr.Zero);
+
+            bool result = Interop.Kernel32.FreeLibrary(handle);
+            if (!result)
+                throw new InvalidOperationException();
+        }
+
+        private static unsafe IntPtr GetSymbolOrNull(IntPtr handle, string symbolName)
+        {
+            var symbolBytes = new byte[Encoding.UTF8.GetByteCount(symbolName) + 1];
```

Comment:
This can use built-in Utf8 marshaller that is more efficient than this.

```
        [LibraryImport(Libraries.Kernel32, StringMarshalling = StringMarshalling.Utf8)]
        internal static partial IntPtr GetProcAddress(IntPtr hModule, string lpProcName);
```